Sources/MongoSwift/ConnectionString.swift

```diff
@@ -297,6 +297,19 @@ internal class ConnectionString {
             try self.setInt32Option(MONGOC_URI_MAXPOOLSIZE, to: value)
         }
 
+        if let connectTimeoutMS = options?.connectTimeoutMS {
+            guard let value = Int32(exactly: connectTimeoutMS), value > 0 else {
```

Comment:
I think 0 is actually a valid value for this option, according to the [URI options spec](https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst#list-of-specified-options):
> non-negative integer; 0 means "no timeout"

relatedly, for a lot of these options we have various tests in `ConnectionStringTests.swift` which verify that we e.g. error on invalid values, correctly prioritize options provided via `MongoClientOptions` over those provided via the connection string itself. I think it would be good to add some of those for this option; you might take some inspiration from e.g. `testLocalThresholdMSOption`